### PR TITLE
Add import/export db for QA. Issue #9766.

### PR DESCRIPTION
**Pull request by gdregalo**
Resolves:
https://github.com/brave/brave-browser/issues/9766 

## Submitter Checklist:

- [x ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by gdregalo** [comment]
Screenshot_20200514-182326

The `publisher_info_db` is exported from application internal folder to `Download\rewards` with timestamp suffix ("Export Rewards db" button).
To import: copy `publisher_info_db` `Download\rewards` (no timestamp suffix) and tap "Import Rewards db" button.

**Comment by NejcZdovc** [comment]
@gdregalo could we add notes to this actions, so that QA knowns where export will go and in which folder and what should be the name when importing? 

Also could be that when you click import you are just prompt to select a file instead of fixed path?

**Comment by gdregalo** [comment]
@NejcZdovc 
Unfortunately Android app can write only in few public folders accessible from outside. I think, the "Download" is the most appropriate one. Choosing a file is possible. But I don't know how this is important for QA.

**Comment by NejcZdovc** [comment]
@gdregalo qa has a lot of test data and if they can select file they can have files names per versions for example so no renaming is needed.

File: android/java/org/chromium/chrome/browser/settings/developer/BraveQAPreferences.java [hunk 1]
```diff
@@ -71,12 +84,40 @@ public void onCreate(Bundle savedInstanceState) {
         if(mDebugNTP != null) {
             mDebugNTP.setOnPreferenceChangeListener(this);
         }
+
+        mDbUtil = BraveDbUtil.getInstance();
+
+        mImportRewardsDb = findPreference(QA_IMPORT_REWARDS_DB);
+        mExportRewardsDb = findPreference(QA_EXPORT_REWARDS_DB);
+        setRewardsDbClickListeners();
+        checkQACode();
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/settings/developer/BraveQAPreferences.java:93]: Moved here for proper `ACTION_GET_CONTENT` handling. When it is `onStart` it request code additionaly, when file is chosen.

**Comment by samartnik** [comment]
In addition to QA options, export option was added to general `Developer options` section to be accessible for users.

device-2020-05-21-073654
device-2020-05-21-073622
device-2020-05-21-073640

**Comment by samartnik** [comment]
Follow up PR for import is created here https://github.com/brave/brave-core/pull/5688

**Comment by srirambv** [comment]
Verification passed on OnePlus 6T with Android 10 running 1.11.28 Nightly build
- Verified able to export Rewards DB via developer options

**Pull request merged**